### ISSUE TITLE
Bug: [@typescript-eslint/no-shadow] False positive between Enum name, and imported function

### ISSUE THREAD

**Issue by Stephen2** [typescript-eslint/typescript-eslint#10749]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.6.2&fileType=.tsx&code=JYWwDg9gTgLgBAbzgOQgOwKZwL5wGZQQhwBEeEEJA3AFA0YAekscGaArsQLIQAmAhgBtENOCnRYAvHADkqTDIA0NbDSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uQAsBDACYcA7uiiJo0DtEjgwAXxDygA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import { None } from "foo";

export enum Modal {
  None = 'None',
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-shadow": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

I expected no error, because the `None` is an Enum value scoped behind `Modal.None` and shouldn't clash with an imported function name

### Actual Result

'None' is already declared in the upper scope on line 1 column 10. 4:3 - 4:7

**Comment by kirkwaiblinger** [comment]
Hi @Stephen2 !

This is working as intended, see https://typescript-eslint.io/rules/no-shadow/#why-does-the-rule-report-on-enum-members-that-share-the-same-name-as-a-variable-in-a-parent-scope

**Comment by Stephen2** [comment]
Is it as simple as declare the Enum in a separate module?

**Comment by kirkwaiblinger** [comment]
You could either do that, yeah, or `import { None as SomeOtherName } from 'foo';`, or else `eslint-disable` that line.... 

Now that you ask the question, though, it does seem like that particular "shadowing" isn't hurting anybody as long as `None` isn't actually _used_ within an enum initializer. I'm wondering if it makes sense to adjust the report only if the shadowed enum declaration is used, in other words

```ts
import { None } from "foo";

// good 
export enum Modal {
  None = 0,
}

// bad
export enum Modal {
  None = 0,
  Some = None + 1,
}
```

@typescript-eslint/triage-team

**Comment by bradzacher** [comment]
> Now that you ask the question, though, it does seem like that particular "shadowing" isn't hurting anybody as long as None isn't actually used within an enum initializer

That's like saying this isn't actually hurting anybody

```ts
import { Foo } from 'module';

namespace Test {
  export const Foo = 1;
}
```

Sure, it's *_technically_* safe cos it shadows but `Foo` isn't _used_ in the scope -- but it's still a shadowing and there's an obvious footgun here. The codebase style is no shadowing allowed...

**Comment by kirkwaiblinger** [comment]
True, true, the only difference in the enum case is that the variable declaration is an involuntary and ignorable side effect of what you're actually trying to do, declare an enum key, rather than the intended effect, as it is in other cases.

**Comment by Stephen2** [comment]
Thanks for considering, I've moved the `enum` declarations to a separate file, and am happy with the result.
